scripts/parse-arguments.sh

```diff
@@ -0,0 +1,82 @@
+#!/usr/bin/env bash
+
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+set -ex
+TEST_ROOT_DIR=$(git rev-parse --show-toplevel)
```

Comment:
A lot of these defaults are repeated again in `pom.xml` - the idea is that maintainers of the framework can run the scripts in isolation without using Maven. This can be done both with the top level `setup-integration-test-env.sh` script, or any of the sub-components - e.g. `clone-spark.sh`.

Reply:
Hmm.. it seems error prone that this needs to stay in sync with the args in pom.xml. Is there a better dispatch mechanism than to use this?

Reply:
The `pom.xml` can assume that these are the defaults and that they are correct, but I like having them explicitly enumerated in the `pom.xml` because the behavior can be known right from the pom. But I also like being able to run the script independently without any arguments. But all of this is just for convenience, so for the sake of developer sanity we can make one of these ideas less convenient.

Don't have a strong opinion on the best way to move forward.